README.md

```diff
@@ -345,47 +346,397 @@ clay export domain.com/_lists > lists.clay
+* `public/css/_kiln-plugins.css` from compiled kiln plugin styles in their `.vue` components
+* `public/js/_global.js` from compiled legacy scripts specified by the `--globs` argument
+* `public/js/<number>.js` from compiled dependencies of `client.js`, `model.js`, kiln plugin, or global scripts
+* `public/js/_deps-<letter>-<letter>.js` from compiled and bundled dependencies
+
+This also creates a number of files that are used for instantiating component controllers, serving scripts, and speeding up incremental builds:
+
+* `public/js/_client-init.js` script that instantiates `client.js` [component controllers which export a default function](https://claycms.gitbook.io/kiln/kiln-fundamentals/components#model-and-controller)
+* `public/js/_prelude.js` dynamic bundler initialization script
+* `public/js/_postlude.js` dynamic bundler access script, adds `window.require()` which enables loading of bundled dependencies
+* `public/js/_ids.json` cache of module IDs, used when serving bundles
+* `public/js/_registry.json` cache of module dependencies, used when serving bundles
+* `client-env.json` environment variables in `model.js` and dependencies, must be added to `.gitignore` (values for these variables are [passed to Amphora via `env`](https://claycms.gitbook.io/amphora/startup/instantiation#instantiation-arguments) on server start)
+* `browserify-cache.json` local cache for fast incremental builds, must be added to `.gitignore`
+
+Specifying `--minify` (or using `CLAYCLI_COMPILE_MINIFIED` or more specifically `CLAYCLI_COMPILE_MINIFIED_SCRIPTS`) will run all compiled scripts through [`terser`](https://github.com/fabiosantoscode/terser).
+
+This will also copy `clay-kiln-edit.js` and `clay-kiln-view.js` to `public/js` if you have Kiln installed. When you specify `--watch`, Kiln scripts will also be watched for changes.
+
+#### Dependency Management
+
+Any files you `require()` or `import` in your `client.js`, `model.js`, kiln plugins, or legacy global JavaScript are compiled to `<number>.js` and `_deps-<letter>-<letter>.js`, based on their name (for example, `lodash` might be compiled to `283.js` and `_deps-i-l.js`). When resolving media, call `claycli.compile.scripts.getDependencies()` in your Clay install's `resolveMedia` function to dynamically load necessary dependencies for view (`client.js` and legacy `_global.js`) and edit (`model.js` and kiln plugins) modes.
+
+By convention, internal services are specified in a `services/` directory at the root of your Clay install. Services that work in both the client and server live in `services/universal/` (or `services/isomorphic/` if you prefer). If you have `services/client/` and `services/server/` directories, `claycli` will automatically substitute server-side dependencies with their client-side equivalents when compiling. This is useful for database / API calls and wrappers around 3rd party libraries that have wildly different Node.js vs browser implementations.
```

Comment:
It's not `convention` technically though, right? Like the `services/` path is hard coded into the logic of this. Semi-colons in JS could a `convention` but in C++ they're required. Which is this path?

Reply:
It's convention because you could put them anywhere (see: `global/js`), but the neato `services/server/` → `services/client/` remapping only happens if you use those two folders